dhis-2/dhis-web/dhis-web-dataentry/src/main/java/org/hisp/dhis/de/action/GetHistoryAction.java

```diff
@@ -114,6 +118,14 @@ public void setUserService( UserService userService )
         this.userService = userService;
     }
 
+    @Autowired
+    private FileResourceService fileResourceService;
+

```

Comment:
You don't need to both Autowire and have a setter for beans. If you autowire it, you don't need to define the resource in the beans.xml, but if you don't use Autowire, you need to add it.